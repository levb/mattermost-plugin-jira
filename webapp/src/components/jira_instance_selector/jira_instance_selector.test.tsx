// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import JiraInstanceSelector from './jira_instance_selector';

describe('components/JiraInstanceSelector', () => {
    const baseActions = {
        getConnected: jest.fn().mockResolvedValue({}),
    };

    const instance1 = {
        instance_id: 'https://something.atlassian.net',
        is_default: true,
        type: 'cloud' as 'cloud',
    };

    const instance2 = {
        instance_id: 'http://localhost:8080',
        is_default: false,
        type: 'server' as 'server',
    };

    const baseProps = {
        ...baseActions,
        onlyShowConnectedInstances: false,
        instances: [instance1, instance2],
        connectedInstances: [instance1],
        theme: {},
        value: '',
        onChange: jest.fn().mockResolvedValue({}),
    };

    test('should match snapshot', () => {
        const props = {...baseProps};
        const wrapper = shallow<JiraInstanceSelector>(
            <JiraInstanceSelector {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when value is set', () => {
        const props = {
            ...baseProps,
            value: instance1.instance_id,
        };
        const wrapper = shallow<JiraInstanceSelector>(
            <JiraInstanceSelector {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when onlyShowConnectedInstances is true', () => {
        const props = {
            ...baseProps,
            onlyShowConnectedInstances: true,
        };
        const wrapper = shallow<JiraInstanceSelector>(
            <JiraInstanceSelector {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
