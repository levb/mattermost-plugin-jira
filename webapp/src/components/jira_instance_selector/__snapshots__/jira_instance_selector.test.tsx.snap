// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/JiraInstanceSelector should match snapshot 1`] = `
<Fragment>
  <ReactSelectSetting
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "https://something.atlassian.net",
          "value": "https://something.atlassian.net",
        },
        Object {
          "label": "http://localhost:8080",
          "value": "http://localhost:8080",
        },
      ]
    }
    theme={Object {}}
  />
</Fragment>
`;

exports[`components/JiraInstanceSelector should match snapshot when onlyShowConnectedInstances is true 1`] = `
<Fragment>
  <ReactSelectSetting
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "https://something.atlassian.net",
          "value": "https://something.atlassian.net",
        },
      ]
    }
    theme={Object {}}
  />
</Fragment>
`;

exports[`components/JiraInstanceSelector should match snapshot when value is set 1`] = `
<Fragment>
  <ReactSelectSetting
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "https://something.atlassian.net",
          "value": "https://something.atlassian.net",
        },
        Object {
          "label": "http://localhost:8080",
          "value": "http://localhost:8080",
        },
      ]
    }
    theme={Object {}}
    value={
      Object {
        "label": "https://something.atlassian.net",
        "value": "https://something.atlassian.net",
      }
    }
  />
</Fragment>
`;
